lua/jdtls/dap.lua

```diff
@@ -429,12 +404,58 @@ function M.setup_dap(opts)
             console = 'integratedTerminal';
           }
           table.insert(configurations, config)
+          if remaining == 0 then
+            callback(configurations)
+          end
         end)
       end)
     end
   end)
 end
 
+local orig_configurations
+function M.setup_dap_main_class_configs()
+  local status, dap = pcall(require, 'dap')
+  if not status then
+    print('nvim-dap is not available')
+    return
+  end
+  if not orig_configurations then
+    orig_configurations = vim.deepcopy(dap.configurations.java)
+  end
+  local current_configurations = vim.deepcopy(orig_configurations)
+  M.fetch_main_configs(function(configurations)
+    vim.list_extend(current_configurations, configurations)
+    dap.configurations.java = current_configurations
+  end)
+end
+
+
+function M.setup_dap(opts)
```

Comment:
I just realised that this could be a breaking/confusing change for existing users. Maybe it would be safer to add an option that disables automatic addition of auto-detected configurations?

Reply:
It's not ideal and so far I  tried to avoid breaking behavior but I think it's still okay - there's still a warning at the top of the readme.